src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -58,6 +58,14 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
                     return;
                 }
 
+                // Don't show up within member access
+                // This previously worked because the type inferrer didn't work
+                // in member access expressions.
+                if (token.IsKind(SyntaxKind.DotToken))
```

Comment:
The implementation has now shuffled around a bunch of times since I started this work, but it appears that we may eventually query every provider without regard to its triggers (probably a bug):

http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionServiceWithProviders.cs,148

and eventually ask all providers for items without regard to their triggers 
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionServiceWithProviders.cs,244